### PR TITLE
fix: fix programId paragraph syntax as per docs.

### PR DESCRIPTION
**Pull request by amanPras**
fix programId paragraph syntax as per docs.
Reference - https://www.ibm.com/docs/en/developer-for-zos/14.1?topic=structure-cobol-program

Making changes in grammar as per the documentation results in slightly different diagnostics for some input. PR has a screenshot of prev state vs the new state.

### CODE REVIEW

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestControlCompilerDirective.java [hunk 1]
```diff
@@ -34,9 +34,9 @@ class TestControlCompilerDirective {
           + "       Program-Id. control-dir.";
 
   private static final String TEXT_WRONG_ARG =
-      "       Identification Division.\n"
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestControlCompilerDirective.java:37]: Previous result
image (1)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestControlCompilerDirective.java [hunk 2]
```diff
@@ -34,9 +34,9 @@ class TestControlCompilerDirective {
...
-          + "       *{CONTROL|1} {MAP1|2}\n"
-          + "       *{CBL|3} MAP1\n"
+      "       Identification Division{.|4}\n"
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestControlCompilerDirective.java:37]: new result
image (2)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java [hunk 3]
```diff
@@ -49,12 +50,13 @@ void test() {
     Map<String, LineRange> expectedRanges =
         new ImmutableMap.Builder<String, LineRange>()
             .put("COPY BAR", new LineRange(0, 0))
-            .put("PROGRAM", new LineRange(1, 5))
-            .put("IDENTIFICATION DIVISION", new LineRange(1, 1))
-            .put("DATA DIVISION", new LineRange(2, 5))
-            .put("WORKING-STORAGE SECTION", new LineRange(3, 5))
-            .put("STRUCTNAME", new LineRange(4, 5))
-            .put("VARNAME", new LineRange(5, 5))
+            .put("PROGRAM: TEST", new LineRange(1, 6))
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java:53]: new result:
image (3)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java [hunk 4]
```diff
@@ -49,12 +50,13 @@ void test() {
     Map<String, LineRange> expectedRanges =
         new ImmutableMap.Builder<String, LineRange>()
             .put("COPY BAR", new LineRange(0, 0))
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java:52]: prev result
image (4)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java (hunk 3 above)
Comment (amanPras) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestOutlineTreeLineNumbers.java:53]: So, added a program-id statement to get the outline

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestSemicolonNotAllowedInIdentificationDivision.java [hunk 5]
```diff
@@ -28,7 +28,7 @@
 class TestSemicolonNotAllowedInIdentificationDivision {
   private static final String TEXT =
       "       IDENTIFICATION DIVISION;\n"
-          + "       {PROGRAM-ID|1}. {TEST1|2}.\n"
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestSemicolonNotAllowedInIdentificationDivision.java:31]: prev :
image (5)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestSemicolonNotAllowedInIdentificationDivision.java [hunk 6]
```diff
@@ -28,7 +28,7 @@
...
+          + "       {PROGRAM-ID|1}{.|2} TEST1.\n"
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestSemicolonNotAllowedInIdentificationDivision.java:31]: new
image (6)

**Review by ishche — APPROVED** [review]

**Pull request merged**